matrix/inverse_of_matrix.py

```diff
@@ -2,47 +2,162 @@
+                + (d(matrix[0][1]) * d(matrix[1][0]) * d(matrix[2][2]))
+                + (d(matrix[0][0]) * d(matrix[1][2]) * d(matrix[2][1]))
+            )
+        )
+        if determinant == 0:
+            raise ValueError("This matrix has no inverse.")
+
+        # Creating cofactor matrix
+        cofactor_matrix = [
+            [d(0.0), d(0.0), d(0.0)],
+            [d(0.0), d(0.0), d(0.0)],
+            [d(0.0), d(0.0), d(0.0)],
+        ]
+        cofactor_matrix[0][0] = (d(matrix[1][1]) * d(matrix[2][2])) - (
+            d(matrix[1][2]) * d(matrix[2][1])
+        )
+        cofactor_matrix[0][1] = -(
+            (d(matrix[1][0]) * d(matrix[2][2])) - (d(matrix[1][2]) * d(matrix[2][0]))
+        )
+        cofactor_matrix[0][2] = (d(matrix[1][0]) * d(matrix[2][1])) - (
+            d(matrix[1][1]) * d(matrix[2][0])
+        )
+        cofactor_matrix[1][0] = -(
+            (d(matrix[0][1]) * d(matrix[2][2])) - (d(matrix[0][2]) * d(matrix[2][1]))
+        )
+        cofactor_matrix[1][1] = (d(matrix[0][0]) * d(matrix[2][2])) - (
+            d(matrix[0][2]) * d(matrix[2][0])
+        )
+        cofactor_matrix[1][2] = -(
+            (d(matrix[0][0]) * d(matrix[2][1])) - (d(matrix[0][1]) * d(matrix[2][0]))
+        )
+        cofactor_matrix[2][0] = (d(matrix[0][1]) * d(matrix[1][2])) - (
+            d(matrix[0][2]) * d(matrix[1][1])
+        )
+        cofactor_matrix[2][1] = -(
+            (d(matrix[0][0]) * d(matrix[1][2])) - (d(matrix[0][2]) * d(matrix[1][0]))
+        )
+        cofactor_matrix[2][2] = (d(matrix[0][0]) * d(matrix[1][1])) - (
+            d(matrix[0][1]) * d(matrix[1][0])
+        )
 
-    # Calculate the determinant of the matrix
-    determinant = d(matrix[0][0]) * d(matrix[1][1]) - d(matrix[1][0]) * d(matrix[0][1])
-    if determinant == 0:
-        raise ValueError("This matrix has no inverse.")
+        # Transpose the cofactor matrix (Adjoint matrix)
+        adjoint_matrix = array(cofactor_matrix)
+        for i in range(3):
+            for j in range(3):
+                adjoint_matrix[i][j] = cofactor_matrix[j][i]
 
-    # Creates a copy of the matrix with swapped positions of the elements
-    swapped_matrix = [[0.0, 0.0], [0.0, 0.0]]
-    swapped_matrix[0][0], swapped_matrix[1][1] = matrix[1][1], matrix[0][0]
-    swapped_matrix[1][0], swapped_matrix[0][1] = -matrix[1][0], -matrix[0][1]
+        # Inverse of the matrix using the formula (1/determinant) * adjoint matrix
+        inverse_matrix = array(cofactor_matrix)
+        for i in range(3):
+            for j in range(3):
+                inverse_matrix[i][j] /= d(determinant)
 
-    # Calculate the inverse of the matrix
-    return [[float(d(n) / determinant) or 0.0 for n in row] for row in swapped_matrix]
+        # Calculate the inverse of the matrix
+        return [[float(d(n)) or 0.0 for n in row] for row in inverse_matrix]
+    else:
+        raise ValueError("Please provide a matrix of size 2x2 or 3x3.")

```

Comment:
```suggestion
    raise ValueError("Please provide a matrix of size 2x2 or 3x3.")
```

Reply:
For me(old school) this change is optional.